userapi/src/main/java/fk/prof/userapi/verticles/HttpVerticle.java

```diff
@@ -69,10 +71,12 @@ private Router configureRouter() {
     public void start(Future<Void> startFuture) throws Exception {
         aggregationWindowDurationInSecs = userapiConfigManager.getAggregationWindowDurationInSecs();
         baseDir = userapiConfigManager.getBaseDir();
+        httpConfig = config().mapTo(Configuration.HttpConfig.class);
```

Comment:
Ya, To perform the validation at startup, it is converted to the class there also. 
But since we are using DeploymentOptions for verticle deployment, its config jsonObject is used to pass around the verticle specific config, and hence must be converted again to use it.